Dockerfile added

coxjonc opened the pull request.

Commit:
Created Dockerfile

Commit:
wqMerge remote-tracking branch 'origin/master'

bmenn commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Do you mind stripping out the Mozilla License comment block and assuming the license on the root directory the project (which is Mozilla License 2.0)? It just to simplify the licensing, so if the license ever need to be updated, we only need to worry about updating one location.

bmenn commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+
+# Install dependencies 
+RUN pip install .
+
```
Can you try removing `/palantiri` after the `pip install .` to keep the Docker image slim?

bmenn commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+RUN pip install .
+
+# Execute test script
+CMD ["/usr/local/bin/python", "setup.py", "test"]
```
I would run the `search.py` script at the root of the repo instead of the tests for the project. If the `CMD` is changed to run `search.py`, don't forget to add the file.

coxjonc replied:
Just to be clear, which file are you referring to? The `--help` output for `search.py` didn't mention the need to specify an output file when you run the script. As far as I can tell the only argument required is a `-b` flag to tell the parser to search Backpage. I haven't had time to look through the source yet so I'm probably just missing something.

coxjonc replied:
Wait scratch that I got it now.

bmenn commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+COPY . /palantiri/
+
+# Make /palantiri our working directory
+WORKDIR /palantiri
```
I personally would not change the `WORKDIR`. If you changed it to help the Pip installation, I think you should be able to do `pip install /palantiri`.

bmenn commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Use the latest python image from DockerHub (3.x)
+FROM python:latest
```
I would specify at least the major version of Python if not also the minor version, so that it is clear in the Dockerfile which Python version you are using.

bmenn reviewed (changes requested):
Nice job figuring out how to get palantiri into a Docker image.

Commit:
Sundry Dockerfile fixes

Removed license, specified python version

Removed /palantiri directory after install

Also installed the lxml parser library for bs4.

coxjonc commented:
Requested changes were made in commit 15e6417
